### PR TITLE
Advertise Test Day

### PR DESCRIPTION
**Pull request by galgeek**
hi @whimboo !

thanks for your feedback on #35 — this patch replaces it, addressing issue #22

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -5,6 +5,8 @@
     "nick": "_TestDayBot",
     "secure": "true",
     "autoRejoin": "true",
-    "testDayAdmins": ["ashughes","whimboo"],
-    "helpers": ["ashughes"]
+    "testDayAdmins": ["ashughes", "whimboo"],
+    "helpers": ["ashughes"],
+    "advertisement": {"channels":["#contributors", "#developers", "#interns", "#introduction", "#mozillians", "#newbies", "#seneca"],
+        "message":"Mozilla QA is holding a Test Day today. Interested in participating? Please join us in #qa!"}
```
Comment (whimboo) [inline config.json]: Please format this properly with the right indentation. There are tools which can assist you if you are unsure.

File: config.json [hunk 2]
```diff
@@ -5,6 +5,8 @@
     "nick": "_TestDayBot",
     "secure": "true",
     "autoRejoin": "true",
-    "testDayAdmins": ["ashughes","whimboo"],
-    "helpers": ["ashughes"]
+    "testDayAdmins": ["ashughes", "whimboo"],
```
Comment (whimboo) [inline config.json]: This is also a change from another PR! Please try to void mixing code changes between PRs.

**Comment by whimboo** [comment]
Beside the code comments please try to get the feedback from Anthony.

**Comment by ashughes1** [comment]
> Not sure where the event organizer will be listed. Means if he/she will be an admin or a helper/moderator. 

I think I'd like the "organizer" to be a separate actor in this case. It would be good at some point to be able to know who is organizing events and not just administrating or helping with an event.

> @ashughes1 might be able to tell us who should be able to send those advertisements. 

Leaving that responsibility to the event organizer should be sufficient for v1.

> We always need fallbacks in case that the organizer is not around regarding time zones.

We're going to be trying out a new model this year where events won't go on for longer than a a couple hours. The timezone issue you speak of should be moot in that model.

**Comment by galgeek** [comment]
hi @ashughes1 and @whimboo

At least for v1, does the "organizer" require different testdaybot privileges from "admins"? 
(It does make sense to me that "organizer" should be responsible for advertising, but I wonder if this needs to be restricted in the code.)

Where do we want to specify the "organizer"? In the "next" command to schedule the Test Day, or "addOrganizer," or ...?

**Comment by ashughes1** [comment]
I don't think an "organizer" needs to have different permissions than an "admin". I would say that an "admin" should be the highest level and an organizer is just a different type of admin.

**Comment by whimboo** [comment]
@ashughes1 given the tight timeline for Barbara would it be ok to first let helpers allow to run the advertising command? Those are set by the admin and are under control for each testday. We could have a follow-up issue to actually implement a real organizer.

**Comment by ashughes1** [comment]
Why not just limit it to event admin then for now?

**Comment by whimboo** [comment]
Admins exist to administrate the bot but not to drive a testday in any given way. Because of that helpers seem to be a way better fit for me as long as we do not have the organizer user level. Also helpers should be trusted. The admin has to set them each time before a testday, so the list is not getting that large.

Also it would be limited to yourself if the admin way is used, given that we most likely don't want to make more people admins beside us two, maybe a fallback for each time zone but that should be all.

**Comment by ashughes1** [comment]
Thanks for clarifying the distinction. I'm fine with Helpers having the power to trigger advertising via the bot.

**Comment by galgeek** [comment]
Hello @whimboo and @ashughes1 

I've updated the `:advertise` code to make it a `helper` command, and otherwise updated the code per Henrik's line notes.

The code is a pretty direct copy from fennecbot, but it doesn't manage to send messages to channels the bot hasn't joined in my testing, while I have the impression that fennecbot does. Is that because the fennecbot code that's in use differs significantly from the github repo, or it that because fennecbot has more privileges than my +opped test bot running in #testdaybotTest? Or am I wrong that fennecbot advertises on multiple channels?

Another issue is what kind of user verification to use when a helper issues the command, or when a new helper is added to the list for a test day.

**Comment by galgeek** [comment]
Hi @whimboo!

The code in this updated PR runs fine with node-irc v0.3.7 in my quick tests. I don't know whether you might want to have a look before the other PRs are landed.

I don't see any similar code in the github repo for the automation bot.

Maybe we should add subcommands to update the message or channels?

Thank you for all your help!

**Comment by galgeek** [comment]
Hi @whimboo!

I've left this code running, using node-irc v0.3.7, in #testdaybotTest.

#qa and #automation are the advertisement channels, and the message is a short test message, "please pardon this test of an update to testdaybot!"

**Comment by whimboo** [comment]
This looks fine now. Given that it is blocked on version 0.3.7 I will defer its merge to default until the IRC plugin has been upgraded.

**Comment by whimboo** [comment]
I squashed all the commits, fixed the initial commit message, and merged the PR as https://github.com/mozilla/testdaybot/commit/f8e779cc6078e0d2d2b58642c6a64e3c356db5cf.

**Pull request closed without merging**